app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java

```diff
@@ -1932,18 +1906,18 @@ public void onReceive(Context context, Intent intent) {
     }
 
     /**
-     * Request edit of image (by image editor/viewer - for example to crop image)
+     * Request edit of file
      *
      * @param file File that should be edited
      */
-    public void requestPictureEdit(final Context context, final File file) {
-        final Uri uri = getUriByFileProviderAuthority(context, file);
+    public void requestFileEdit(final Context context, final File file) {
+        final Uri uri = FileProvider.getUriForFile(context, getFileProvider(context), file);
         final Intent intent = new Intent(Intent.ACTION_EDIT);
-        intent.setDataAndType(uri, "image/*");
-        intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION | Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        intent.setData(uri);  // Mime type inferred automatically
+        intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
         intent.putExtra(MediaStore.EXTRA_OUTPUT, uri);
-        intent.putExtra(EXTRA_FILEPATH, file.getAbsolutePath());
```

Comment:
Should be fine to keep sending this bit of info? For some software (like Markor 😄 ) it makes it easier for them to work if they have the full real path.

Reply:
Added it back in